routing/router.go

```diff
@@ -1292,6 +1292,21 @@ func (r *ChannelRouter) processUpdate(msg interface{},
 		channelID := lnwire.NewShortChanIDFromInt(msg.ChannelID)
 		fundingTx, err := r.fetchFundingTx(&channelID)
 		if err != nil {
+			// If the funding transaction isn't found at all, then
+			// we'll mark the edge itself as a zombie so we don't
+			// continue to request it.
+			//
+			// TODO(roasbeef): mark as zero if strict?
+			zErr := r.cfg.Graph.MarkEdgeZombie(
+				msg.ChannelID, msg.NodeKey1Bytes,
+				msg.NodeKey2Bytes,
```

Comment:
Is it safe to mark zero pubkeys here even without running in strict mode? The channel doesn't have a valid UTXO backing it, so it could never be resurrected anyway.

Reply:
What strict mode started to do was modify the values stored here to enforce that the non-blank pubkey was used here for resurrection: https://github.com/lightningnetwork/lnd/pull/5138/commits/f5cf6a0b70965de82c5d9947e0d075cdfefdc03b. So this should be fine IIUC

Reply:
Let's mark both keys as zero then? Saves space (66 bytes per entry) and makes sure we never resurrect them.

Reply:
We'll still end up writing a `[33]byte{}` so no bytes, saved, but moved to using blank keys so they can't be resurrected.

Reply:
Actually, what about the rare case where for w/e reason the RPCs in the pipeline to fetch the funding transaction or UTXO return an error, which means we'll never be able to properly re-validate the channel (assuming it isn't actually spent)?

Reply:
Gonna add some new error types so we can isolate the failures.

Reply:
If someone ends a like bad block hash (by way of an invalid short channel ID), it'll still be added to the reject cache so we'll only process it a single time within that restart cycle.

Reply:
Dug a bit and it'll really only fail when there's something wrong on the node side, added another error case to catch failed funding validation.